### PR TITLE
Add funpark01 - Take #3

### PR DESCRIPTION
**Pull request by abcdefg30**
Next try after #5320 and #6792.

Three things I need a decision on:
- [x] The map still contains unused/empty areas. What about having trees/civilians/builidngs/dinos there?
- [x] https://github.com/OpenRA/OpenRA/pull/8085#discussion_r29555610
- [x] Do we really want the dinos to walk this (ugly) way?

### CODE REVIEW

File: mods/cnc/maps/funpark01/scj01ea.lua [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	DinoTric.Patrol({WP0.Location, WP1.Location}, true, 3)
+	DinoTrex.Patrol({WP2.Location, WP3.Location}, false)
+	Trigger.OnIdle(DinoTrex, DinoTrex.Hunt)
+
+	ReinforceWithLandingCraft(RifleReinforcments, SeaEntryA.Location, BeachReinforceA.Location, BeachReinforceA.Location)
+	Trigger.AfterDelay(DateTime.Seconds(1), function() InitialUnitsArrived = true end)
+
+	Trigger.AfterDelay(DateTime.Seconds(15), function() ReinforceWithLandingCraft(BazookaReinforcments, SeaEntryB.Location, BeachReinforceB.Location, BeachReinforceB.Location) end)
+	--Trigger.AfterDelay(DateTime.Seconds(200), function() ReinforceWithLandingCraft(BikeReinforcments, SeaEntryA.Location, BeachReinforceA.Location, BeachReinforceA.Location) end)
+	--Trigger.AfterDelay(DateTime.Seconds(220), function() ReinforceWithLandingCraft(BikeReinforcments, SeaEntryB.Location, BeachReinforceB.Location, BeachReinforceB.Location) end)
```
Comment (abcdefg30) [inline mods/cnc/maps/funpark01/scj01ea.lua]: So, what do to with this?
```yaml
I see three options:
a) Remove those two lines
b) Send those reinforcements when your initial units are killed
c) Send them earlier
```

I'm for `a)`. IMHO the mission plays fine without extra reinforcements.

Comment (Mailaender) [inline mods/cnc/maps/funpark01/scj01ea.lua]: d) Tie it to an easy difficulty.

**Comment by dan9550** [comment]
The Dinos walked in that "ugly" way in the original, you could say it adds some comic flare haha

I'm against changing things in the empty areas, change for the sake of change.

**Comment by abcdefg30** [comment]
All open points solved.

File: mods/cnc/maps/funpark01/scj01ea.lua [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	transport.Wait(5)
+	transport.ScriptedMove(transportStart)
+	transport.Destroy()
+
+	Media.PlaySpeechNotification(player, "Reinforce")
+end
+
+initialSong = "j1"
+PlayMusic = function()
```
Comment (Mailaender) [inline mods/cnc/maps/funpark01/scj01ea.lua]: This function is never triggered.

File: mods/cnc/maps/funpark01/scj01ea.lua [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+	Trigger.OnPlayerWon(nod, function()
+		Media.PlaySpeechNotification(nod, "Win")
+	end)
+
+	Trigger.OnPlayerLost(nod, function()
+		Media.PlaySpeechNotification(nod, "Lose")
+	end)
+
+	ReachVillageObj = nod.AddPrimaryObjective("Reach the village.")
```
Comment (Mailaender) [inline mods/cnc/maps/funpark01/scj01ea.lua]: Seems a bit redundant to `InvestigateObj`.

**Comment by Mailaender** [comment]
`TREX` works as promised. Only the music left to fix.

File: mods/cnc/maps/funpark01/scj01ea.lua (hunk 2 above)
Comment (abcdefg30) [inline mods/cnc/maps/funpark01/scj01ea.lua]: Ops, seems like it swallowed that somewhere.

File: mods/cnc/maps/funpark01/scj01ea.lua (hunk 3 above)
Comment (abcdefg30) [inline mods/cnc/maps/funpark01/scj01ea.lua]: I added it to make the players task more clear (he has to search for the village anyway).
Do you want me to remove it?

Comment (pchote) [inline mods/cnc/maps/funpark01/scj01ea.lua]: IMO this is already implied by "Investigate the nearby village", so it can be removed.

**Comment by abcdefg30** [comment]
Fixed the music.

**Comment by obrakmann** [comment]
Very close to the original, [only our dinos move somewhat slower](https://youtu.be/3YZSMHVV0T8). Code and yaml look good, too. The jungle terrain is a nice touch. :+1:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/995c7d4f5e113c8e00166c1a7773987b5d1c8978...fb75be3ae0480834d19568fd4a2ab5a9fbaba8fe)

**Comment by Mailaender** [comment]
Just in time. www.jurassicworld.com :palm_tree:

**Pull request merged**